package-toolkit/runtime/src/main/kotlin/com/atlan/pkg/Utils.kt

```diff
@@ -765,6 +765,36 @@ object Utils {
             (!directUpload && objectStoreKey.isNotBlank())
     }
 
+    /**
+     * Get the list of files in the provided directory.
+     *
+     * @param importType details of the type of import expected (direct vs object store-based)
+     * @param directory path to the directory to list files from
+     * @return list of absolute paths of files in the directory
+     */
+    fun getFilesInDirectory(
+        importType: String,
+        directory: String,
+    ): List<String> {
+        if (importType == "DIRECT") {
+            val path = Paths.get(directory)
+            val files = mutableListOf<String>()
+            if (path.isDirectory()) {
+                path.walk().forEach {
+                    if (it.isRegularFile()) {
+                        files.add(it.pathString)
+                    }
+                }
+            } else {
+                files.add(directory)
+            }
+            return files
+        } else {
+            // return empty list for object store-based imports
+            return listOf()
```

Comment:
I expect someone providing multiple files via an object store is going to be the most common scenario (even with a zip file for the upload widget, there is a file size limit there that is quite small — only ~10 MB). So here we'd really need to provide some way of walking all objects in the object store under a particular prefix (i.e. if the input provided contains only a prefix, and no object key, we should get all object keys in that prefix).